### ISSUE TITLE
Multiprocess forking does not work on osx

### ISSUE THREAD

**Issue by xoviat** [pyinstaller/pyinstaller#2759]
This was introduced in 7f63111f4f19d4815b6a3235b6d14122e8f9510d. We probably need to disable the spawn fix for forking.

**Comment by xoviat** [comment]
Related #2322.

**Comment by mkassner** [comment]
@xoviat   Instead of the current fix in pyinstaller, I m using this solution by @bpteague:

https://gist.github.com/bpteague/bc574e4d92c9d9904ceba895fb694fc8


It works perfectly without side-effects.

**Comment by xoviat** [comment]
The commit is the patch with some minor integration adjustments. It apparently does not work perfectly.

**Comment by mkassner** [comment]
Ah ok. I m running this on Mac with bundles that mulitprocess in  forking and non-forking manner.

**Comment by xoviat** [comment]
FYI [test failure is](https://travis-ci.org/pyinstaller/pyinstaller/jobs/263153179#L4237)

```
[8253] LOADER: Running pyi_multiprocess_forking.py
Fatal Python error: Cannot recover from stack overflow.
Current thread 0x00007fff78b41000 (most recent call first):
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
  File "PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py", line 53 in __init__
```

**Comment by htgoebel** [comment]
Huh, that's curious: Line 53 is:
```
    class _Popen(forking.Popen):
        def __init__(self, *args, **kw):
                …
                super(_Popen, self).__init__(*args, **kw)
```
Which would mean `pyi_rth_multiprocessing._Popen` is it's own super-class.

**Comment by springermac** [comment]
@htgoebel I would consider this a blocker for 3.3 as it breaks existing functionality.

**Comment by xoviat** [comment]
@springermac What is the way forward on this issue?

2017-08-31 18:12 GMT-05:00 Jonathan Springer <notifications@github.com>:

**Comment by htgoebel** [comment]
I agree this is a heavy issue. But is anybody working on it? Is there an idea how to solve it? I want to release 3.3 ASAP (maybe even this weekend), so I'd rather have a known issue then to wait for weeks until this one is solved.

**Comment by bpteague** [comment]
Fixed by #2795
